### DISCUSSION TITLE
days_count

### DISCUSSION THREAD

**Discussion by craigmediaservices** [bug-or-feature/bc-utils#14] [General]
Where are you finding days_count for instruments? I was adding some instruments to the config and can't figure out where you're sourcing the information. Thank you!

**Comment by bug-or-feature** [comment]
Finger in the air. Ultimately I suppose from an approximation the RollOffsetDays in roll config in PST. In fact that would probably be a better number to use for futures. My own use case was for an entirely different setup. There are probably some instruments where there should be a value set, and it's missing, and others where the value is not high enough

**Comment by BorderCollieLover** [comment]
My two cents: 

Practically, one can use a large number. BarChart limits only the number of datasets downloaded each day (250), but for each dataset, you can download up to 20000 data points.  In my fork, I pass end_date to _before_available_res to ensure at least some data is available for a contract. 

I also looked at the MRCI  (https://www.mrci.com/web/index.php) dataset as a proxy for trading interests/liquidity. 75% of all contracts will have meaningful data of less than 400 days, so I use 400 as default.  Only a handful of assets have much longer trading periods than that, most notably a few fixed income instruments, SONIA, SOFR, EDOLLAR, 3-M EURO-Yen, and things like GOLD, CORN, SOYBEAN.

**Reply by littlebigfis** [reply]
Are there any instances you can think of where someone might need more data than whats afforded in bcutils. I'm thinking of contracts like EDOLLAR with large RollOffsetDays. Maybe I'm just confusing myself?

**Comment by bug-or-feature** [comment]
Thanks @BorderCollieLover I have created a reminder to allow the days_count to be configurable in the next release

**Comment by bug-or-feature** [comment]
This is done. default_day_count is 400, but you can override it per call to `get_barchart_downloads()`
